snap/seed_yaml.go

```diff
@@ -68,6 +68,18 @@ func ReadSeedYaml(fn string) (*Seed, error) {
 		return nil, fmt.Errorf("cannot unmarshal %q: %s", yamlData, err)
 	}
 
+	// XXX: filter-out nil snaps
+	// For context see: https://bugs.launchpad.net/snapd/+bug/1825437
+	filtered := make([]*SeedSnap, 0, len(seed.Snaps))
+	for _, sn := range seed.Snaps {
+		if sn != nil {
+			filtered = append(filtered, sn)
+		}
+	}
+	if len(filtered) != len(seed.Snaps) {
+		seed.Snaps = filtered
+	}
+
```

Comment:
I think this is the wrong approach. Yes we should not assume it's going to be valid, but if it's not valid (and an entry being nil is just one way for it not to be valid), we should ignore the whole thing (and log the fact, loudly).

Reply:
@chipaca I agree in principle but not in this case. There is no loud logging that users will see. Users cannot fix that. There is only the yet-undiscovered issue that is affecting some ISOs that causes an otherwise valid YAML to have a single NULL entry.

The real bummer is that this should never have happened. The cleanup now should be focused on ensuring that broken installations can recover but IMO the only way to do that is with a snapd.deb update.